lib/curl_ntlm_core.c

```diff
@@ -143,14 +143,10 @@ static void setup_des_key(const unsigned char *key_56,
   DES_cblock key;
 
   /* Expand the 56-bit key to 64-bits */
-  extend_key_56_to_64(key_56, (char *) key);
+  extend_key_56_to_64(key_56, (char *) &key);
```

Comment:
DES_cblock is a struct wrapping an array in BoringSSL and a plain array in OpenSSL. (Array types in C act really surprisingly, even when typedef'd. I wish OpenSSL hadn't done that.) Sometimes code needs the occasional `&` to smooth it over. :-/ On the plus side, it makes it clearer that you're mutating the key.